tools/generator/src/Extensions/XCScheme+Extensions.swift

```diff
@@ -0,0 +1,127 @@
+             buildableReference = .init(
+                 pbxTarget: pbxTarget,
+                 referencedContainer: referencedContainer
+             )
+        }
+    }
+}
+
+// MARK: XCScheme.BuildableReference
+
+extension XCScheme.BuildableReference {
+    convenience init(pbxTarget: PBXTarget, referencedContainer: String) {
+        self.init(
+            referencedContainer: referencedContainer,
+            blueprint: pbxTarget,
+            buildableName: pbxTarget.buildableName,
+            blueprintName: pbxTarget.name
+        )
+    }
+}
+
+// MARK: XCScheme.BuildAction
+
+extension XCScheme.BuildAction {
+    convenience init<PBXTargetInfos: Sequence>(
+        buildMode: BuildMode,
+        targetInfos: PBXTargetInfos,
+        hostBuildableReference: XCScheme.BuildableReference?,
+        hostIndex: Int?
+    ) where PBXTargetInfos.Element == XCScheme.PBXTargetInfo {
+        let buildableReferences = targetInfos.map(\.buildableReference)
+
+        let entries: [XCScheme.BuildAction.Entry] = (buildableReferences + [hostBuildableReference])
+            .compactMap { $0 }
+            .map { .init(withDefaults: $0) }
+
+        let preActions: [XCScheme.ExecutionAction]
+        if buildMode.usesBazelModeBuildScripts {
+            let targetInfoPreActions: [XCScheme.ExecutionAction] = targetInfos
+                .compactMap { .init(targetInfo: $0, hostIndex: hostIndex) }
+            preActions = targetInfoPreActions.isEmpty ? [] :
+                [.clearBazelBuildOutputGroupsFile] + targetInfoPreActions
+        } else {
+            preActions = []
+        }
+
+        self.init(
+            buildActionEntries: entries,
+            preActions: preActions,
+            parallelizeBuild: true,
+            buildImplicitDependencies: true
+        )
+    }
+}
+
+// MARK: XCScheme.BuildAction.Entry
+
+extension XCScheme.BuildAction.Entry {
+    convenience init(withDefaults buildableReference: XCScheme.BuildableReference) {
+        self.init(
+            buildableReference: buildableReference,
+            buildFor: [
+                .running,
+                .testing,
+                .profiling,
+                .archiving,
+                .analyzing,
+            ]
+        )
+    }
+}
+
+// MARK: XCScheme.ExecutionAction
+
+extension XCScheme.ExecutionAction {
+    static let clearBazelBuildOutputGroupsFile = XCScheme.ExecutionAction(
+        scriptText: #"""
+if [[ -s "$BAZEL_BUILD_OUTPUT_GROUPS_FILE" ]]; then
+    rm "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
+fi
+"""#,
+        title: "Clear Bazel Build Output Groups File"
+    )
+
+    convenience init?(targetInfo: XCScheme.PBXTargetInfo, hostIndex: Int?) {
+        guard targetInfo.pbxTarget is PBXNativeTarget else {
+          return nil
+        }
+
+        let hostTargetOutputGroup: String
+        if let hostIndex = hostIndex {
+            hostTargetOutputGroup = #"""
+echo "b $BAZEL_HOST_TARGET_ID_\#(hostIndex)" >> "$BAZEL_BUILD_OUTPUT_GROUPS_FILE"
+"""#
```

Comment:
@brentleyjones For a scheme, is it true that the host target output group only needs to appear once in the `$BAZEL_BUILD_OUTPUT_GROUPS_FILE`? The logic that currently exists adds it for every buildable target which does not seem right to me.

Reply:
Correct. It only needs to appear once, but it doesn't harm it to appear multiple times, and the logic to try it prevent it from appearing multiple times is most likely too much of a hassle compared to just blindly doing it this way.